lib/ip_record.js

```diff
@@ -41,15 +41,16 @@ module.exports = function (limits, now) {
     // IPs are limited based on the number of unique email
     // addresses they access.  Take the highest-weighted
     // bad-login event for each email address.
-    var total = 0
-    var seen = {}
+    var weights = {}
     this.lf.forEach(function(info) {
-      var incr = limits.badLoginErrnoWeights[info.e] || 1
-      if (info.u in seen && seen[info.u] < incr) {
-        total -= seen[info.u]
-        seen[info.u] = incr
-      }
-      total += incr
```

Comment:
The previous code here was trying to be smart and keep a running total by subtracting the old weight and adding the new one, but it's buggy.  So I replaced it with two separate loops that more obviously do the right thing.